packages/monitor-v2/src/llm-bot/OptimisticOracleV2Client.ts

```diff
@@ -0,0 +1,113 @@
+import { Provider } from "@ethersproject/abstract-provider";
+import { paginatedEventQuery } from "@uma/common";
+import { RequestPriceEvent } from "@uma/contracts-frontend/dist/typechain/core/ethers/OptimisticOracleV2";
+import { OptimisticOracleV2Ethers } from "@uma/contracts-node";
+import { Event, EventFilter, ethers } from "ethers";
+import { blockDefaults } from "../utils/constants";
+import { getContractInstanceWithProvider, tryHexToUtf8String } from "../utils/contracts";
+import {
+  BlockRange,
+  OptimisticOracleClient,
+  OptimisticOracleRequest,
+  OptimisticOracleType,
+  calculateRequestId,
+} from "./common";
+
+export class OptimisticOracleClientV2 extends OptimisticOracleClient<OptimisticOracleRequest> {
+  constructor(
+    _provider: Provider,
+    _requests: Map<string, OptimisticOracleRequest> = new Map<string, OptimisticOracleRequest>(),
+    _fetchedBlockRanges?: BlockRange[]
```

Comment:
whats the purpose of this and where is it checked or updated?

Reply:
This helps us track the block ranges that have been updated/fetch so we can control that there are no gaps/overlaps

It's only updated in the constructor: https://github.com/UMAprotocol/protocol/blob/b04e25776ab034129b389317e9e005aaeeb3cb24/packages/monitor-v2/src/llm-bot/common.ts#L164

Checks are done here: https://github.com/UMAprotocol/protocol/blob/b04e25776ab034129b389317e9e005aaeeb3cb24/packages/monitor-v2/src/llm-bot/common.ts#L225